kpc/views.py

```diff
@@ -1,3 +1,93 @@
+from django_datatables_view.base_datatable_view import BaseDatatableView
+from django.contrib import messages
+from django.urls import reverse_lazy
+from django.contrib.auth.mixins import LoginRequiredMixin, UserPassesTestMixin
 
-# Create your views here.
+
+from .forms import CertificateRegisterForm
+from .models import Certificate, DeliveryStatus
+from django.contrib.auth import get_user_model
+
+User = get_user_model()
+
+
+@permission_required('accounts.can_get_licensee_contacts', raise_exception=True)
+def licensee_contacts(request):
+    """Return users associated with the provided licensee"""
+    if request.method == 'GET':
+        licensee_id = request.GET.get('licensee',)
+        contacts = User.objects.filter(profile__licensees=licensee_id) if licensee_id else []
+        if contacts:
+            contacts_json = [{'id': user.id, 'name': user.get_full_name()} for user in contacts]
+        else:
+            contacts_json = []
+    else:
+        return HttpResponse(status=405)
+    return JsonResponse(contacts_json, safe=False)
+
+
+class CertificateRegisterView(LoginRequiredMixin, UserPassesTestMixin, FormView):
+    raise_exception = True
+
+    def test_func(self):
+        """only for superusers"""
+        if self.request.user.is_superuser:
+            return True
+
+    template_name = 'certificate/register.html'
+    form_class = CertificateRegisterForm
+    success_url = reverse_lazy('cert-register')
+
+    def form_valid(self, form):
+        """Generate requested Certificates"""
+        method = form.cleaned_data['registration_method']
+        cert_kwargs = {'assignor': self.request.user, 'licensee': form.cleaned_data['licensee'],
+                       'date_of_issue': form.cleaned_data['date_of_issue'],
+                       'status': DeliveryStatus.objects.get(slug='assigned')}
+        if method == 'sequential':
+            start = form.cleaned_data['cert_from']
+            end = form.cleaned_data['cert_to']
+            certs = [i for i in range(start, end+1)]
+        if method == 'list':
+            certs = [cert_number.strip()[2:] for cert_number in form.cleaned_data['cert_list'].split(',')]
+
+        if certs:
+            Certificate.objects.bulk_create([Certificate(number=i, **cert_kwargs) for i in certs])
+
+        messages.success(self.request, f'Generated {len(certs)} new certificates.')
+        return super().form_valid(form)
+
+
+class CertificateJson(LoginRequiredMixin, BaseDatatableView):
+    model = Certificate
+    columns = ['number', 'licensee', 'status', 'date_of_issue', 'last_modified']
+    order_columns = columns
+
+    max_display_length = 500
+
+    def get_initial_queryset(self):
+        """Limit to certificates visible to user"""
+        return self.request.user.profile.certificates()
+
+    def filter_queryset(self, qs):
+        # Filter by certificate number
+        search = self.request.GET.get(u'search[value]', None)
```

Comment:
No need for `u` here -- we're all unicode all the time in Python3 :)

Also, I'm pretty sure `.get`'s default if the entry isn't present is `None`, so that might be redundant here.